junit-platform-console/src/main/java/org/junit/platform/console/tasks/TreePrintingListener.java

```diff
@@ -0,0 +1,226 @@
+		this.verticals[2] = ""; // "engine" level
+		for (int i = 3; i < verticals.length; i++) {
+			verticals[i] = verticals[i - 1] + theme.vertical();
+		}
+	}
+
+	@Override
+	public void testPlanExecutionStarted(TestPlan testPlan) {
+		frames.push(new Frame(testPlan.toString()));
+	}
+
+	@Override
+	public void testPlanExecutionFinished(TestPlan testPlan) {
+		frames.pop();
+	}
+
+	@Override
+	public void executionStarted(TestIdentifier testIdentifier) {
+		executionStartedNanos = System.nanoTime();
+		if (testIdentifier.isContainer()) {
+			printVerticals(theme.entry());
+			printf(BLUE, " %s", testIdentifier.getDisplayName());
+			printf(NONE, "%n");
+			frames.push(new Frame(testIdentifier.getUniqueId()));
+		}
+	}
+
+	@Override
+	public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult testExecutionResult) {
+		if (testIdentifier.isContainer()) {
+			frames.pop();
+			return;
+		}
+		Color color = Color.valueOf(testExecutionResult);
+		printVerticals(theme.entry());
+		printf(NONE, " %s", testIdentifier.getDisplayName());
+		// printf(NONE, " %d ms", durationInMillis(System.nanoTime() - executionStartedNanos));
+		printf(color, " %s", theme.computeStatusTile(testExecutionResult));
+		testExecutionResult.getThrowable().ifPresent(t -> printf(color, " %s", t.getMessage()));
+		printf(NONE, "%n");
+	}
+
+	@Override
+	public void executionSkipped(TestIdentifier testIdentifier, String reason) {
+		printVerticals(theme.entry());
+		printf(testIdentifier.isContainer() ? BLUE : NONE, " %s", testIdentifier.getDisplayName());
+		printf(YELLOW, " %s %s%n", theme.skipped(), reason);
+	}
+
+	void printf(Color color, String message, Object... args) {
+		if (monochrome || color == NONE) {
+			out.printf(message, args);
+		}
+		else {
+			// Use string concatenation to avoid ANSI disruption on console
+			out.printf(color + message + NONE, args);
+		}
+		out.flush();
+	}
+
+	/** Print verticals and stay in the current line. */
+	void printVerticals(String tile) {
+		printf(NONE, verticals[Math.min(frames.size(), verticals.length)]);
+		printf(NONE, tile);
+	}
+
+	long durationInMillis(long nanos) {
+		return TimeUnit.NANOSECONDS.toMillis(nanos);
+	}
+
+	enum Theme {
+		/**
+		 * ASCII 7-bit characters form the tree branch.
+		 *
+		 * <pre>
+		 * .
+		 * +-- JUnit Vintage
+		 * |  +-- example.JUnit4Tests
+		 * |  |  -  standardJUnit4Test
+		 * +-- JUnit Jupiter
+		 * |  +-- AssertionsDemo
+		 * |  |  -  timeoutExceeded()
+		 * |  |  -  exceptionTesting()
+		 * </pre>
+		 */
+		ASCII(".", "| ", "+--", "---", "[OK]", "[A]", "[X]", "[S]"),
+
+		/**
+		 * Extended ASCII characters are used to display the test execution tree.
+		 *
+		 * <pre>
+		 * .
+		 * ├─ JUnit Vintage
+		 * │  ├─ example.JUnit4Tests
+		 * │  │  ├─ standardJUnit4Test
+		 * ├─ JUnit Jupiter
+		 * │  ├─ A stack
+		 * │  │  ├─ is instantiated with new Stack()
+		 * </pre>
+		 */
+		UTF_8(".", "│  ", "├─", "└─", "✔", "■", "✘", "⏩");
```

Comment:
Hm, the `fast forward` (skipped) glyph rendered by a browser does not meet the look-and-feel of the other three: `"✔", "■", "✘"`. Change it to another glyph?